### DISCUSSION TITLE
[10.x] Frontend search

### DISCUSSION THREAD

**Discussion by akno77** [pimcore/pimcore#10246] [Q&A / Support]
Hi,

I'm wondering if there is any documentation or sample how to implement a frontend search (pages, data objects like news etc.) in Pimcore 10? In the demo there is only a sample for ecommerce product search...

Also all the external search bundles doesn't seem to work with Pimcore 10...

Why isn't there a simple frontend search available through Pimcore by default?

**Comment by dpfaffenbauer** [comment]
Because there is no generic solution for a frontend search. I would suggest dachcom-digital/dynamic-search, but that currently doesn't work for Pimcore X.

**Comment by beheist** [comment]
Hey @akno77 ,
I had the same issue and monkey-patched the LuceneSearch package to work with Pimcore X. Not everything works yet (e.g. Admin Controller is still broken), but the indexer runs and you can render search results:
https://github.com/SHE-IT/pimcore-lucene-search

You can install it by adding the following entry to your composer.json:
```
"repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/SHE-IT/pimcore-lucene-search"
        }
    ]
```
and then just running `composer require dachcom-digital/lucene-search:dev-master`.
No guarantees though, I did this last night and have not really tested much yet. ;)

**Reply by beheist** [reply]
Once I tested some more and fixed the broken things that I can find, I will submit a PR back to upstream.

**Reply by akno77** [reply]
Thanks, I will try it!

**Reply by akno77** [reply]
@beheist Crawling doesn't work for me, I get the exception:

```
[Symfony\Component\ErrorHandler\Error\UndefinedFunctionError]`       
  Attempted to call function "str" from namespace "GuzzleHttp\Psr7".  
                                                                      
Exception trace:
  at /var/www/html/vendor/vdb/php-spider/src/Resource.php:78
```
Also including the result action in my template like this:

`{{ render(controller('LuceneSearchBundle:List:getResult')) }}`
 Causes an error:

`An exception has been thrown during the rendering of a template ("The controller for URI "/_fragment" is not callable: Controller "LuceneSearchBundle:List:getResult" does neither exist as service nor as class.").`

If I change the call to:
`{{ render(controller('LuceneSearchBundle\\Controller\\ListController:getResultAction')) }}`

I get the error:

`An exception has been thrown during the rendering of a template ("You have requested a non-existent service "pimcore.templating". Did you mean one of these: "pimcore.templating.engine.delegating", "pimcore.templating.engine.twig"?").`

**Comment by kovinet** [comment]
Do we have any working solutions (easy to implement) site search for Pimcore in 2025?

**Reply by kovinet** [reply]
https://github.com/dachcom-digital/pimcore-dynamic-search looks solid.

**Reply by dpfaffenbauer** [reply]
this is what we use and recommend as well
